ui/src/app/shared/header/chips/datepicker-input.component.ts

```diff
@@ -0,0 +1,32 @@
+import {Component, EventEmitter, Input, OnInit, Output} from "@angular/core";
```

Comment:
Doesn't have to be in this PR, but one nice future improvement might be to auto-open the datepicker when you open the mat menu.

Reply:
Done. I think this is a more legitimate use of setTimeout since we're waiting on a timed UI animation rather than for lifecycle code to execute, and the calendar doesn't position properly before the dropdown is finished redrawing for the last time. I'm on the fence about this, so if you think this is messy, we can remove the timeout and keep the behavior as is (calendar opens immediately before card expansion animation is complete and overlays the card instead of positioning below it).

Reply:
I tested this using menuOpened() but that callback seems to also trigger before the animation completes on the menu.